apps/publisher/extensions/assets/default/apis/assets.jag

```diff
@@ -43,170 +93,143 @@ require('/modules/publisher.js').exec(function (ctx) {
-                                        options.type);
+                    if (!contentTypeheader) {
+                        response = responseProcessor.buildErrorResponseDefault(constants.STATUS_CODES.UNSUPPORTED_MEDIATYPE, 'error on create asset', response, 'Unsupported Media Type of creating ' + options.type, 'more information on create asset error', []);
+                    }
+                    if (jsonHeader >= 0 || multipartHeader >= 0 || formHeader >= 0) {
+                        addAsset(options, req, res, session);
                     }
+                    response = responseProcessor.buildErrorResponseDefault(constants.STATUS_CODES.UNSUPPORTED_MEDIATYPE, 'error on create asset', response, 'Unsupported Media Type of creating ' + options.type, 'more information on create asset error', []);
                 } catch (e) {
-                    log.error(e);
-                    response = responseProcessor.buildErrorResponse(res,
-                            constants.STATUS_CODES.INTERNAL_SERVER_ERROR, 'Failed to create asset of type: ' +
-                                    options.type);
+                    log.error(e.message, e);
+                    response = responseProcessor.buildErrorResponseDefault(constants.STATUS_CODES.INTERNAL_SERVER_ERROR, 'error on create asset', response, 'Failed to create asset of type: ' + options.type, 'more information on create asset error', []);
                 }
-
             } else if (uriMatcher.match(UPDATE_URL)) {//to update asset
+                contentTypeheader = req.getHeader('Content-Type');
+                jsonHeader = contentTypeheader.indexOf('application/json');
+                multipartHeader = contentTypeheader.indexOf('multipart/form-data');
+                formHeader = contentTypeheader.indexOf('application/x-www-form-urlencoded');
                 options.id = uriMatcher.elements().id;
                 try {
-                    asset = assetApi.update(options, req, res, session);
-                    //if the asset is successfully updated
-                    if (asset) {
-                        response = responseProcessor.buildSuccessResponse(response,
-                                constants.STATUS_CODES.OK, asset)
-                    } else {
-                        response = responseProcessor.buildErrorResponse(response,
-                                constants.STATUS_CODES.INTERNAL_SERVER_ERROR, 'Unable to update the asset');
+                    if (!contentTypeheader) {
+                        response = responseProcessor.buildErrorResponseDefault(constants.STATUS_CODES.UNSUPPORTED_MEDIATYPE, 'error on create asset', response, 'Unsupported Media Type of creating ' + options.type, 'more information on create asset error', []);
+                    }
+                    if (jsonHeader >= 0 || multipartHeader >= 0 || formHeader >= 0) {
+                        updateAsset(options, req, res, session);
```

Comment:
How does updateAsset knows to read the data. i.e. Depending on the content type, the way you read the data would be different. Hence, you will have to have content type header check in updateAsset() method as well, which is redundant. Hence, organize as below.

if(jsonHeader) {
    updateAssetJSON();
    return;
}
if(multipartHeader) {
    updateAssetMultipart();
    return;
}

....